platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/GenesisStateBuilder.java

```diff
@@ -68,11 +64,18 @@ public static ReservedSignedState buildGenesisState(
             @NonNull final MerkleRoot stateRoot) {
 
         final BasicConfig basicConfig = platformContext.getConfiguration().getConfigData(BasicConfig.class);
-        stateRoot.setPlatformState(buildGenesisPlatformState(addressBook, appVersion));
+        final PlatformStateAccessor platformState;
+        if (stateRoot instanceof MerkleStateRoot root) {
+            platformState = new PlatformStateAccessorSingleton(root);
+        } else {
+            platformState = new PlatformState();
+        }
```

Comment:
What about adding a method to `MerkleRoot` like `initPlatformState()` or `createPlatformState()` to avoid instanceof checks here?

Reply:
Actually, I realized that this code is not even needed. I can remove it entirely.